crates/services/sync/src/import.rs

```diff
@@ -170,7 +181,11 @@ where
             let p2p = p2p.clone();
```

Comment:
We need to modify things a little further - and remove the double buffering going on here. The original issue was that the first buffer will only be polled once the second buffer gets drained.

For the initial v2 approach, I simply combined fetching the header and the `get_block_for_header` logic into the same `tokio::spawn` task. This gave a big improvement to sync delays, however if we want to bring back the double buffer approach we'll need to fixup the v4 approach as we discussed at some later date.

Reply:
I see! With the introduction of the extra header streams and stream of streams, I was thinking it might mean a double buffer. But I will take another look at that redesign now.